src/app/Http/Controllers/BusinessTripController.php

```diff
@@ -35,46 +51,236 @@ public function create() {
+        if (!file_exists($filePath)) {
+            abort(404, 'File not found'); // Or other error handling
+        }
+
+        // Download the file
+        return response()->download($filePath, $trip->upload_name);
     }
 
     /**
      * Returning the view with the trip editing form
+     * @throws \Exception
      */
-    public function edit(BusinessTrip $b) {
-        return view();
+    public function edit(BusinessTrip $trip) {
+        $startDate = new DateTime($trip->datetime_start);
+        $endDate = new DateTime($trip->datetime_end);
+        $interval = new DateInterval('P1D');
+        $dateRange = new DatePeriod($startDate, $interval, $endDate);
+
+        $days = iterator_count($dateRange);
+
+        // Get the collection of contributions associated with the trip
+        $tripContributions = $trip->contributions()->pluck('contributions.id');
+
+        return view('business-trips.edit',  [
+            'trip' => $trip,
+            'days' => $days,
+            'tripContributions' => $tripContributions,
+        ]);
+
     }
 
     /**
-     * Validation as in create(), should check if user is admin/not
      * Redirecting to the trip editing form
      * Sending mail with mail component to admin
      */
-    public function update(Request $request, BusinessTrip $b) {
+    public function update(Request $request, BusinessTrip $trip) {
+        //Validate data
+        $validatedData = $request->validate([
+            'user_id' => 'required|integer|min:0',
+            'country_id' => 'required|exists:countries,id',
+            'transport_id' => 'required|exists:transports,id',
+            'place' => 'required|string|max:200',
+            'event_url' => 'nullable|string|max:200',
+            'upload_name' => 'nullable|string|max:200',
+            'sofia_id' => 'nullable|string|max:40',
+            'state' => 'required|integer',
+            'datetime_start' => 'required|date',
+            'datetime_end' => 'required|date|after:datetime_start',
+            'place_start' => 'nullable|string|max:200',
+            'place_end' => 'nullable|string|max:200',
+            'datetime_border_crossing_start' => 'nullable|date',
+            'datetime_border_crossing_end' => 'nullable|date',
+            'trip_purpose_id' => 'required|integer|min:0', //possibly |exists:trip_purposes,id
+            'purpose_details' => 'nullable|string|max:50',
+            'iban' => 'required|string|max:34',
+            'conference_fee_id' => 'nullable|integer|min:0',
+            'reimbursement_id' => 'nullable|integer|min:0',
+            'spp_symbol_id' => 'nullable|exists:spp_symbols,id',
+            'accommodation_expense_id' => 'nullable|integer|min:0',
+            'travelling_expense_id' => 'nullable|integer|min:0',
+            'other_expense_id' => 'nullable|integer|min:0',
+            'advance_expense_id' => 'nullable|integer|min:0',
+            'not_reimbursed_meals' => 'nullable|string|max:255',
```

Comment:
taketo pole vo formulari nie je, je tam cela velka tabulka, teda ju treba prejst a zapisat si hodnoty. nazvy poli najdes u mna v edit view, ale teda volaju sa vzdy b{n}, l{n}, d{n} pre kazdy riadok. kedze su to checkboxy, musis sa pozriet, ktore z tychto mien sa v requeste nachadzaju a pre ne zapisat 1, pre ostatne 0. a az to bude nase 'not_reimbursed_meals' hodnota.